Test.cpp

```diff
@@ -142,11 +142,25 @@ void testReceiveTimeout()
     end = std::chrono::steady_clock::now();
     dur = std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count();
 
-    TEST_EQUALS(m2->getMsgId(), PolyM::MSG_TIMEOUT);
+    TEST_EQUALS(m2.get(), (PolyM::Msg*) nullptr);
```

Comment:
Is the cast necessary?

Reply:
Curiously enough, my compiler on my Fedora 33 is not making the template substitution, leading to a compilation error